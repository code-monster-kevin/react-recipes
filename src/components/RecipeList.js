import React from 'react';
import PropTypes from 'prop-types';
import RecipeListItem from './RecipeListItem';

const RecipeList = ({ style, favourites, recipes, ...props }) => (
  <ul className="list-reset">
    {recipes.map(recipe => (
      <RecipeListItem
        key={recipe.id}
        recipe={recipe}
        favourited={favourites.includes(recipe.id)}
        {...props}
      />
    ))}
  </ul>
);

RecipeList.propTypes = {
  recipes: PropTypes.array,
  favourites: PropTypes.array,
  style: PropTypes.object,
};

RecipeList.defaultProps = {
  recipes: [],
  favourites: [],
};

export default RecipeList;
