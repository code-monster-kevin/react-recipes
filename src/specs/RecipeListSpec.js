import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../components/RecipeList';

const mockRecipes = [
  {
    id: 1,
    name: 'Mock Recipe 1',
    category: 'Mock Category 1',
  },
  {
    id: 2,
    name: 'Mock Recipe 2',
    category: 'Mock Category 2',
  },
  {
    id: 3,
    name: 'Mock Recipe 3',
    category: 'Mock Category 3',
  },
  {
    id: 4,
    name: 'Mock Recipe 4',
    category: 'Mock Category 4',
  },
];

describe('<RecipeList />', () => {
  test('No recipes props input', () => {
    const component = renderer.create(<RecipeList />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Valid recipes props input', () => {
    const component = renderer.create(<RecipeList recipes={mockRecipes} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Render favourite state', () => {
    const component = renderer.create(
      <RecipeList recipes={mockRecipes} favourites={[1, 4]} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
