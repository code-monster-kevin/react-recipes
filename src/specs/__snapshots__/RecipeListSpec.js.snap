// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> No recipes props input 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<RecipeList /> Render favourite state 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="Favourite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ‚ù§Ô∏è
    </span>
    <span>
      Mock Recipe 1
    </span>
    <span>
      Mock Category 1
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="Favourite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      üñ§
    </span>
    <span>
      Mock Recipe 2
    </span>
    <span>
      Mock Category 2
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="Favourite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      üñ§
    </span>
    <span>
      Mock Recipe 3
    </span>
    <span>
      Mock Category 3
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="Favourite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      ‚ù§Ô∏è
    </span>
    <span>
      Mock Recipe 4
    </span>
    <span>
      Mock Category 4
    </span>
  </li>
</ul>
`;

exports[`<RecipeList /> Valid recipes props input 1`] = `
<ul
  className="list-reset"
>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="Favourite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      üñ§
    </span>
    <span>
      Mock Recipe 1
    </span>
    <span>
      Mock Category 1
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="Favourite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      üñ§
    </span>
    <span>
      Mock Recipe 2
    </span>
    <span>
      Mock Category 2
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="Favourite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      üñ§
    </span>
    <span>
      Mock Recipe 3
    </span>
    <span>
      Mock Category 3
    </span>
  </li>
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={[Function]}
  >
    <span
      aria-label="Favourite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      üñ§
    </span>
    <span>
      Mock Recipe 4
    </span>
    <span>
      Mock Category 4
    </span>
  </li>
</ul>
`;
